pkg/api/v1alpha1/awsiamconfig_webhook.go

```diff
@@ -78,7 +78,7 @@ func validateImmutableAWSIamFields(new, old *AWSIamConfig) field.ErrorList {
 	if !new.Spec.Equal(&old.Spec) {
 		allErrs = append(
 			allErrs,
-			field.Invalid(field.NewPath("spec", "AWSIamConfig"), new, "config is immutable"),
+			field.Forbidden(field.NewPath("spec", "AWSIamConfig", new.Name), "config is immutable"),
```

Comment:
I wonder if this is better? However we should see what capi does as an example
```suggestion
			field.Forbidden(field.NewPath("AWSIamConfig", new.Name), "spec is immutable"),
```